Tools/automation.ps1

```diff
@@ -0,0 +1,790 @@
+
+    .EXAMPLE
+    .\automation.ps1 -ResourcePrefix "contoso-" -Index "Demo" -AzResourceGroup "WinGet_PrivateRepo_Demo"
+
+    .EXAMPLE
+    .\automation.ps1 -ResourcePrefix "contoso-" -Index "001" -AzResourceGroup "WinGet_PrivateRepo" -AzLocation "westus"
+
+    .EXAMPLE
+    .\automation.ps1 -ResourcePrefix "contoso-" -Index "Prod" -AzResourceGroup "WinGet_PrivateRepo_Prod" -AzSubscription "Contoso Global" -ShowConnectionInstructions
+
+    .NOTES
+    Something
+    #>
+
+param(
+    [Parameter(Position=0, Mandatory=$true)]  [string]$ResourcePrefix,
+    [Parameter(Position=1, Mandatory=$true)]  [string]$Index,
+    [Parameter(Position=2, Mandatory=$true)]  [string]$AzResourceGroup,
+    [Parameter(Position=3, Mandatory=$false)] [string]$AzSubscriptionName,
+    [Parameter(Position=4, Mandatory=$false)] [string]$AzLocation         = "westus",
+    [Parameter(Position=5, Mandatory=$false)] [string]$WorkingDirectory   = "$($(Get-Item $PSScriptRoot).fullname)\src\WinGet.RestSource.Infrastructure",
+    [Parameter(Position=6, Mandatory=$false)] [string]$ArchiveFunctionZip = "CompiledFunctions.zip"
+)
+
+Function Test-RequiredModules
+{
+    Param(
+        [Parameter(Position=0, Mandatory=$true)] [string]$RequiredModule
+    )
+    Begin
+    {}
+    Process
+    {
+        ## Determinds if the PowerShell Module is missing, If missing Returns the name of the missing module
+        IF(!$(Get-Module -ListAvailable -Name $RequiredModule) )
+            { $Result = $RequiredModule }
+    }
+    End
+    {
+        ## Returns a value only if the module is missing
+        Return $Result
+    }
+}
+
+Function New-WinGetRepo
+{
+    param(
+        [Parameter(Position=0)] [string]$ResourcePrefix,
+        [Parameter(Position=1)] [string]$Index,
+        [Parameter(Position=2)] [string]$AzResourceGroup,
+        [Parameter(Position=3)] [string]$AzSubscriptionName,
+        [Parameter(Position=4)] [string]$AzLocation,
+        [Parameter(Position=5)] [string]$WorkingDirectory = $PSScriptRoot,
+        [Parameter(Position=5)] [string]$ArchiveFunctionZip,
+        [Parameter(Position=6)] [switch]$ShowConnectionInstructions
+    )
+    Begin
+    {
+        $ParameterFolderPath = "$WorkingDirectory\Parameters"       # Path that will be used to target the Parameter files.
+        $TemplateFolderPath  = "$WorkingDirectory\Templates"        # Path that will be used to target the ARM Template files.
+        $RequiredModules     = @("Az.Resources", "Az.Accounts", "Az.KeyVault","Az.Websites", "Az.Functions")
+        $ArchiveFunctionZip  = "$WorkingDirectory\$ArchiveFunctionZip"
+    }
+    Process
+    {
+        ## Test that the Required Azure Modules are installed
+        $Result = @()
+        foreach( $RequiredModule in $RequiredModules )
+            { $Result += Test-RequiredModules -RequiredModule $RequiredModule }
+        
+        IF( $Result )
```

Comment:
if vs IF. We should strive to be consistent with our non-case sensitive keywords throughout the script.

Id generally use all lowercase.

Reply:
I can fix that.

Reply:
Fixed, aligned the IF, Else, and ForEach